server-spi-private/src/main/java/org/keycloak/organization/OrganizationProvider.java

```diff
@@ -56,10 +57,26 @@ public interface OrganizationProvider extends Provider {
     OrganizationModel getByDomainName(String domainName);
 
     /**
-     * Returns the organizations of the given realm as a stream.
-     * @return Stream of the organizations. Never returns {@code null}.
+     * Returns all organizations in the realm.
+     *
+     * @return a {@link Stream} of the realm's organizations.
+     */
+    default Stream<OrganizationModel> getOrganizations() {
+        return getOrganizations("", null, -1, -1);
+    }
+
+    /**
+     * Returns the organizations in the realm using the specified filters.
+     *
+     * @param search a {@code String} representing either an organization name or domain.
+     * @param exact if {@code true}, the organizations will be searched using exact match for the {@code search} param - i.e.
+     *              either the organization name or one of its domains must match exactly the {@code search} param. If false,
+     *              the method returns all organizations whose name or (domains) partially match the {@code search} param.
+     * @param first index of the first element (pagination offset).
+     * @param max the maximum number of results.
+     * @return a {@link Stream} of the matched organizations. Never returns {@code null}.
      */
-    Stream<OrganizationModel> getAllStream();
+    Stream<OrganizationModel> getOrganizations(String search, Boolean exact, Integer first, Integer max);
```

Comment:
I would either rename to `search` or use `getAll`.

Even though I don't link much the format we have in other places, we are also appending `Stream` to the end of the method name when returning streams.